validphys2/src/validphys/coredata.py

```diff
@@ -299,6 +299,96 @@ def with_central_value(self, cv):
          tb = self.commondata_table.copy()
          tb["data"] = cv
          return dataclasses.replace(self, commondata_table=tb)
+    
+    def with_MULT_sys(self,mult_sys):
+        """
+        returns a CommonData instance with MULT systematics
+        replaced by mult_sys
+
+        Parameters
+        ----------
+        mult_sys : pd.DataFrame()
+                 all MULT columns of CommonData.commondata_table
+        """
+        table = self.commondata_table.copy()
+        table["MULT"] = mult_sys
+        return dataclasses.replace(self, commondata_table = table)
+    
+    def with_ADD_sys(self,add_sys):
+        """
+        returns a CommonData instance with ADD systematics
+        replaced by add_sys
+
+        Parameters
+        ----------
+        add_sys : pd.DataFrame()
+                 all ADD columns of CommonData.commondata_table
+        """
+        table = self.commondata_table.copy()
+        table["ADD"] = add_sys
+        return dataclasses.replace(self, commondata_table = table)
+
+    def multiplicative_errors_rescale(self,CORR,UNCORR,sys_rescaling_factor):
+        """
+        rescale the MULT sys by constant factor, sys_rescaling_factor,
+        a distinction is done between CORR and UNCORR systematics
+
+        Parameters
+        ----------
+        CORR : bool
+        UNCORR : bool
+        sys_rescaling_factor : float, int
+
+        Returns
+        -------
+        pd.DataFrame corresponding to the rescaled MULT systematics
+        """
+        mult_table = self.systematics_table.loc[:,["MULT"]].copy()
+        # get indices of CORR / UNCORR sys
+        mult_systype_corr = self.systype_table[(self.systype_table["type"] == "MULT") 
+                            & (~self.systype_table["name"].isin(["UNCORR","THEORYUNCORR"]))]
+        
+        mult_systype_uncorr = self.systype_table[(self.systype_table["type"] == "MULT") 
+                            & (self.systype_table["name"].isin(["UNCORR","THEORYUNCORR"]))]
+
+        # rescale systematics
+        if CORR:
+            mult_table.iloc[:,mult_systype_corr.index - 1] *= sys_rescaling_factor
+        if UNCORR:
+            mult_table.iloc[:,mult_systype_uncorr.index - 1] *= sys_rescaling_factor
+
+        return mult_table
+    
+    def additive_errors_rescale(self,CORR,UNCORR,sys_rescaling_factor):
```

Comment:
If I have understood correctly, this function is exactly the same as `multiplicative_errors_rescale` with the only difference being the string "MULT" in place of "ADD" in `self.systematics_table.loc[:,["MULT"]].copy()`. So I believe we can have a single function with an additional argument that can be either "MULT" or "ADD" and that's it. Do you agree?

Reply:
Thanks for this @andreab1997, I completely agree. The updated version should do so.
`with_MULT_sys` and `with_ADD_sys` are probably redundant as well. I'll take a look at that too asap.